src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private StringBuilder stringBuilder = new StringBuilder();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFromFile(fromFileName);
+        String report = createReport(data);
+        File toFile = new File(toFileName);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFile))) {
+            bufferedWriter.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to the file " + toFile, e);
+        }
+    }
+
+    private String readFromFile(String fileName) {
+        File fromFile = new File(fileName);
+        stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file " + fromFile, e);
+        }
+        return stringBuilder.toString();
```

Comment:
Don't complicate your solution with redundant logic. Also, you can use `Files.readAllLines`
```suggestion
    private List<String> readFromFile(String fileName) {
        File fromFile = new File(fileName);
       List<String> lines = new ArrayList();
        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile))) {
            String value = bufferedReader.readLine();
            while (value != null) {
                lines.add(value);
                value = bufferedReader.readLine();
            }
        } catch (IOException e) {
            throw new RuntimeException("Can't read data from the file " + fromFile, e);
        }
        return lines;
```